### ISSUE TITLE
ymmh values zeroed on signal return

### ISSUE THREAD

**Issue by derekbruening** [DynamoRIO/dynamorio#3812]
The app's ymmh values are being zeroed out somewhere between handle_sigreturn and post-sigreturn dispatch.  In a proprietary app I am observing this with custom diagnostics:

```
signal interruption sigcontext:
	xmm1=0x3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e
mcontext pre-delivery:
	zmm1=0x3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e0000000000000000000000000000000000000000000000000000000000000000
about to sigreturn to this sigcontext:
	xmm1=0x3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e3f7fe40e
resuming 0x000000004b818f40 post-sigreturn:
	zmm1=0x3f7fe40e3f7fe40e3f7fe40e3f7fe40e000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
```

It leads to a crash.

Augmenting api.state_detach to test ymmh (unfortunately I bailed on that when first writing it) we see the same problem on detach:

```
$ ctest -V -R detach_state
293: Test timeout computed to be: 1500
293: ERROR: detach changed ymmh0.hi from 0x1122334455667788 to 0x0
293: ERROR: detach changed ymmh0.lo from 0x11223344556677 to 0x0
293: ERROR: detach changed ymmh0.hi from 0x1122334455667788 to 0x0
293: ERROR: detach changed ymmh0.lo from 0x11223344556677 to 0x0
293: all done
1/1 Test #293: code_api|api.detach_state ........***Failed  Required regular expression not found.Regex=[^all done
```

**Comment by derekbruening** [comment]
The bug is that DR fails to preserve the second magic word at the end of the signal frame xstate.  The word is kind of a "floating" trailing word not present as a field in a struct, making it easily overlooked.

What the kernel does if it fails to find the trailing magic word is that it only restores the x87 and SSE state, and initializes (i.e., zeroes) all the AVX+ state.

**Comment by hgreving2304** [comment]
We had fixed the magic1 value in dde70b1682d49837e6f37. My understanding was that the magic2 value is for sanity checks only. In any case should have re-visited the magic2 value at the same time. Did above commit break it?

**Comment by derekbruening** [comment]
> Did above commit break it?

No, it was required though since missing the magic1 value should result in the exact same symptoms.
